testing/code/test_excinfo.py

```diff
@@ -238,7 +238,10 @@ def f(n):
                 n += 1
             f(n)
 
+        # Restore trace function (https://bugs.python.org/issue36474).
+        oldtrace = sys.gettrace()
         excinfo = pytest.raises(RuntimeError, f, 8)
+        sys.settrace(oldtrace)
```

Comment:
Do you think it makes sense to let `pytest.raises` handle this, maybe optionally by default only?

Reply:
The problem here is that Python forgets about the trace function with RecursionErrors (https://bugs.python.org/issue36474).